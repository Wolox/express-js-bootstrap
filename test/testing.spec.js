const utils = require('./helpers/utils'),
  { mockCommand } = require('./helpers/mocks'),
  { basicFiles, examplePrompts, testingFiles } = require('./helpers/constants');

beforeAll(() => mockCommand());

const ciOptions = ['mocha-chai', 'jest-supertest'];

describe.each(ciOptions)('%s project', testing => {
  beforeAll(() =>
    utils.runKickoff({
      ...examplePrompts,
      projectName: 'TestingProject',
      testing
    })
  );

  test(`creates files for ${testing} project`, () => {
    utils.checkExistentFiles([basicFiles, testingFiles], 'TestingProject');
  });

  test.each(testingFiles)('creates expected %s', file => {
    expect(utils.getFileContent(`TestingProject/${file}`)).toMatchSnapshot();
  });
});
