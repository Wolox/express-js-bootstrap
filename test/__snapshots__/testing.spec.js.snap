// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-supertest project creates expected package.json 1`] = `
"{
  \\"name\\": \\"TestingProject\\",
  \\"version\\": \\"0.1.0\\",
  \\"description\\": \\"Example\\",
  \\"engines\\": {
    \\"node\\": \\"8.9.12\\",
    \\"npm\\": \\"6.4.1\\"
  },
  \\"scripts\\": {
    \\"console\\": \\"node console.js\\",
    \\"cover\\": \\"NODE_ENV=testing jest --coverage --runInBand --forceExit test/app\\",
    \\"test\\": \\"NODE_ENV=testing jest test/app --runInBand --forceExit\\",
    \\"test-inspect\\": \\"NODE_ENV=testing node --inspect --debug-brk jest test/app.spec.js\\",
    \\"eslint-check\\": \\"eslint --print-config .eslintrc.js | eslint-config-prettier-check\\",
    \\"lint\\": \\"eslint \\\\\\"**/*.js\\\\\\"\\",
    \\"lint-diff\\": \\"git diff --name-only --cached --relative | grep \\\\\\\\\\\\\\\\.js$ | xargs eslint\\",
    \\"lint-fix\\": \\"eslint \\\\\\"**/*.js\\\\\\" --fix\\",
    \\"precommit\\": \\"npm run lint-diff\\",
    \\"outdated\\": \\"npm outdated --depth 0\\",
    \\"pretest\\": \\"npm run lint\\",
    \\"prestart\\": \\"npm run lint\\",
    \\"start\\": \\"nodemon --inspect server.js\\"
  },
  \\"cacheDirectories\\": [
    \\"node_modules\\"
  ],
  \\"main\\": \\"app.js\\",
  \\"author\\": \\"Wolox\\",
  \\"homepage\\": \\"https://test.com.ar\\",
  \\"license\\": \\"MIT\\",
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"https://test.com.ar.git\\"
  },
  \\"bugs\\": {
    \\"url\\": \\"https://test.com.ar/issues\\",
    \\"email\\": \\"tls@wolox.com.ar\\"
  },
  \\"jest\\": {
    \\"collectCoverageFrom\\": [
      \\"**/*.js\\",
      \\"!**/console.js\\",
      \\"!**/node_modules/**\\",
      \\"!**/build/**\\",
      \\"!**/migrations/**\\",
      \\"!**/config/**\\",
      \\"!**/scripts/**\\",
      \\"!**/docs/**\\"
    ],
    \\"testEnvironment\\": \\"node\\",
    \\"transform\\": {
      \\"^.+\\\\\\\\.js$\\": \\"babel-jest\\"
    }
  },
  \\"dependencies\\": {
    \\"bcryptjs\\": \\"^2.4.3\\",
    \\"body-parser\\": \\"^1.18.2\\",
    \\"express\\": \\"^4.16.2\\",
    \\"jwt-simple\\": \\"^0.5.1\\",
    \\"morgan\\": \\"^1.9.1\\",
    \\"umzug\\": \\"^2.1.0\\",
    \\"winston\\": \\"^2.4.0\\",
    \\"winston-daily-rotate-file\\": \\"^1.7.2\\"
  },
  \\"devDependencies\\": {
    \\"babel\\": \\"6.23.0\\",
    \\"babel-core\\": \\"6.26.0\\",
    \\"babel-eslint\\": \\"^8.2.2\\",
    \\"babel-jest\\": \\"^23.6.0\\",
    \\"jest\\": \\"^24.0.0\\",
    \\"supertest\\": \\"^3.4.2\\",
    \\"babel-preset-es2015\\": \\"6.24.1\\",
    \\"dictum.js\\": \\"^1.0.0\\",
    \\"dotenv\\": \\"^5.0.0\\",
    \\"eslint\\": \\"^4.8.0\\",
    \\"eslint-config-airbnb-base\\": \\"^12.0.2\\",
    \\"eslint-config-prettier\\": \\"^2.3.1\\",
    \\"eslint-plugin-import\\": \\"^2.6.1\\",
    \\"eslint-plugin-prettier\\": \\"^2.1.1\\",
    \\"husky\\": \\"^0.14.3\\",
    \\"istanbul\\": \\"^0.4.3\\",
    \\"mocha\\": \\"^5.0.1\\",
    \\"mocha-lcov-reporter\\": \\"^1.2.0\\",
    \\"nodemon\\": \\"^1.18.7\\",
    \\"prettier\\": \\"^1.8.2\\",
    \\"prettier-eslint\\": \\"^8.2.1\\",
    \\"prompt\\": \\"^1.0.0\\"
  }
}
"
`;

exports[`jest-supertest project creates expected test/app.spec.js 1`] = `
"'use strict';

const fs = require('fs'),
  path = require('path');


// including all test files
const normalizedPath = path.join(__dirname, '.');

const requireAllTestFiles = pathToSearch => {
  fs.readdirSync(pathToSearch).forEach(file => {
    if (fs.lstatSync(\`\${pathToSearch}/\${file}\`).isDirectory()) {
      requireAllTestFiles(\`\${pathToSearch}/\${file}\`);
    } else {
      require(\`\${pathToSearch}/\${file}\`);
    }
  });
};

requireAllTestFiles(normalizedPath);
"
`;

exports[`mocha-chai project creates expected package.json 1`] = `
"{
  \\"name\\": \\"TestingProject\\",
  \\"version\\": \\"0.1.0\\",
  \\"description\\": \\"Example\\",
  \\"engines\\": {
    \\"node\\": \\"8.9.12\\",
    \\"npm\\": \\"6.4.1\\"
  },
  \\"scripts\\": {
    \\"console\\": \\"node console.js\\",
    \\"cover\\": \\"NODE_ENV=testing istanbul cover ./node_modules/mocha/bin/_mocha  test/app.spec.js\\",
    \\"test\\": \\"NODE_ENV=testing ./node_modules/mocha/bin/_mocha --timeout 6000 --exit test/app.spec.js\\",
    \\"test-inspect\\": \\"NODE_ENV=testing node --inspect --debug-brk ./node_modules/mocha/bin/_mocha test/app.spec.js\\",
    \\"eslint-check\\": \\"eslint --print-config .eslintrc.js | eslint-config-prettier-check\\",
    \\"lint\\": \\"eslint \\\\\\"**/*.js\\\\\\"\\",
    \\"lint-diff\\": \\"git diff --name-only --cached --relative | grep \\\\\\\\\\\\\\\\.js$ | xargs eslint\\",
    \\"lint-fix\\": \\"eslint \\\\\\"**/*.js\\\\\\" --fix\\",
    \\"precommit\\": \\"npm run lint-diff\\",
    \\"outdated\\": \\"npm outdated --depth 0\\",
    \\"pretest\\": \\"npm run lint\\",
    \\"prestart\\": \\"npm run lint\\",
    \\"start\\": \\"nodemon --inspect server.js\\"
  },
  \\"cacheDirectories\\": [
    \\"node_modules\\"
  ],
  \\"main\\": \\"app.js\\",
  \\"author\\": \\"Wolox\\",
  \\"homepage\\": \\"https://test.com.ar\\",
  \\"license\\": \\"MIT\\",
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"https://test.com.ar.git\\"
  },
  \\"bugs\\": {
    \\"url\\": \\"https://test.com.ar/issues\\",
    \\"email\\": \\"tls@wolox.com.ar\\"
  },
  \\"dependencies\\": {
    \\"bcryptjs\\": \\"^2.4.3\\",
    \\"body-parser\\": \\"^1.18.2\\",
    \\"express\\": \\"^4.16.2\\",
    \\"jwt-simple\\": \\"^0.5.1\\",
    \\"morgan\\": \\"^1.9.1\\",
    \\"umzug\\": \\"^2.1.0\\",
    \\"winston\\": \\"^2.4.0\\",
    \\"winston-daily-rotate-file\\": \\"^1.7.2\\"
  },
  \\"devDependencies\\": {
    \\"babel\\": \\"6.23.0\\",
    \\"babel-core\\": \\"6.26.0\\",
    \\"babel-eslint\\": \\"^8.2.2\\",
    \\"babel-preset-es2015\\": \\"6.24.1\\",
    \\"chai\\": \\"^4.1.2\\",
    \\"chai-http\\": \\"^4.2.0\\",
    \\"dictum.js\\": \\"^1.0.0\\",
    \\"dotenv\\": \\"^5.0.0\\",
    \\"eslint\\": \\"^4.8.0\\",
    \\"eslint-config-airbnb-base\\": \\"^12.0.2\\",
    \\"eslint-config-prettier\\": \\"^2.3.1\\",
    \\"eslint-plugin-import\\": \\"^2.6.1\\",
    \\"eslint-plugin-prettier\\": \\"^2.1.1\\",
    \\"husky\\": \\"^0.14.3\\",
    \\"istanbul\\": \\"^0.4.3\\",
    \\"mocha\\": \\"^5.0.1\\",
    \\"mocha-lcov-reporter\\": \\"^1.2.0\\",
    \\"nodemon\\": \\"^1.18.7\\",
    \\"prettier\\": \\"^1.8.2\\",
    \\"prettier-eslint\\": \\"^8.2.1\\",
    \\"prompt\\": \\"^1.0.0\\"
  }
}
"
`;

exports[`mocha-chai project creates expected test/app.spec.js 1`] = `
"'use strict';

const fs = require('fs'),
  chaiHttp = require('chai-http'),
  chai = require('chai'),
  path = require('path');

chai.use(chaiHttp);

// including all test files
const normalizedPath = path.join(__dirname, '.');

const requireAllTestFiles = pathToSearch => {
  fs.readdirSync(pathToSearch).forEach(file => {
    if (fs.lstatSync(\`\${pathToSearch}/\${file}\`).isDirectory()) {
      requireAllTestFiles(\`\${pathToSearch}/\${file}\`);
    } else {
      require(\`\${pathToSearch}/\${file}\`);
    }
  });
};

requireAllTestFiles(normalizedPath);
"
`;
